cle/backends/java/apk.py

```diff
@@ -73,6 +78,78 @@ def __init__(self, apk_path, binary_stream, entry_point=None, entry_point_params
                                   jni_libs_ld_path=jni_libs_ld_path,
                                   **options)
 
+        # the lifecycle needs to support of pyaxmlparser
+        if PYAXMLPARSER_INSTALLED:
+            self.components = {'activity': [], 'service': [], 'receiver': [], 'provider': []}
+            self.callbacks = {'activity': [], 'service': [], 'receiver': [], 'provider': []}
+            self._set_lifecycle(apk_parser)
+
+    def _set_lifecycle(self, apk_parser):
+        """
+        Set components with callbacks of APK lifecycle.
+
+        :param pyaxmlparser apk_parser: XML Parser of the APK.
+        """
+
+        component_getter = {'activity': apk_parser.get_activities,
+                            'service': apk_parser.get_services,
+                            'receiver': apk_parser.get_receivers,
+                            'provider': apk_parser.get_providers}
+
+        for key, getter in component_getter.items():
+            class_names = getter()
+            self.components[key], self.callbacks[key] = self._extract_lifecycle(class_names, key)
+
+    def _extract_lifecycle(self, cls_name: list, component_kind: str) -> (list, list):
```

Comment:
Don't use `list`. Use `List` from `typing` instead. That way you can specify the types of elements inside the list: `cls_name: List[str]`.

Reply:
```list``` is replaced to List. I import ```SootMethod``` and ```SootClass``` for the type annotation of List.